packages/storefront-lwc/src/modules/commerce/productDetail/productDetail.js

```diff
@@ -30,7 +31,11 @@ export default class ProductDetail extends LightningElement {
         selectedColor: '',
     };
 
-    @api set pid(val) {
+    @wire(routeParams) params(params) {
```

Comment:
what does it do? Do we need to do this for basketapi too?

Reply:
This wires in the route parameter for the product id. It is provided by the router. `routeParams` is a wire adapter imported from `@lwce/router`